server/src/databases/cosmosClient.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  CosmosClient as AzureCosmosClient,
+  CosmosClientOptions,
+  ContainerRequest,
+  Container,
+  Database
+} from '@azure/cosmos';
+import { DefaultAzureCredential } from '@azure/identity';
+import { ServerConfigModel } from '../models/configModel';
+
+class CosmosClient extends AzureCosmosClient {
+  private _cosmosDBName: string;
+  private _database: Database | undefined;
+  private _container: Container | undefined;
+
+  constructor({ cosmosDBConnectionString, cosmosDBEndpoint, cosmosDBName }: ServerConfigModel) {
+    const cosmosClientOptions: CosmosClientOptions = {
+      endpoint: cosmosDBEndpoint,
+      aadCredentials: new DefaultAzureCredential()
+    };
+
+    if (cosmosDBConnectionString) {
```

Comment:
I think we should also check for emptiness here, to handle the case where conn string is empty but endpoint is defined. That is to say, default to using the endpoint if connstring is not valid.
`if (cosmosDBConnectionString && cosmosDBConnectionString.length > 0)`